library/Icingadb/Model/Behavior/ReRoute.php

```diff
@@ -39,6 +46,21 @@ public function rewriteCondition(Filter\Condition $condition, $relation = null)
                 );
             }
 
+            if (
+                in_array(substr($relation, 0, -1), self::MIXED_TYPE_RELATIONS)
+                && strpos($remainingPath, 'servicegroup.') !== false
```

Comment:
Please use substr here and make a startswith check. I don't like the fact that `servicegroup.` may appear anywhere.